bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1442,7 +1443,7 @@ private void normalizeRuleConfigurations(Rule rule) {
 
     }
 
-    private <T extends Module> void normalizeModuleConfigurations(List<T> modules) {
+    private <T extends Module> void normalizeModuleConfigurations(List<@NonNull T> modules) {
```

Comment:
If we state that the list must not contain an entry that is equal to null, the Eclipse nullness checker knows that the elements on iterating are non-null, too.
And IMHO it is also good to inform the caller that the list must not contain a null entry.